tests/test_npe2.py

```diff
@@ -164,6 +164,53 @@ def test_valid_mutations(mutator, uses_sample_plugin):
     PluginManifest(**data)
 
 
+@pytest.mark.parametrize(
+    "display_name",
+    [
+        "Here there everywhere and more with giggles and friends",
+        "ab",
+        " abc",
+        "abc ",
+        "_abc",
+        "abc_",
+        "abc♱",
+    ],
+)
+def test_invalid_display_names(display_name, uses_sample_plugin):
+    pm = list(PluginManifest.discover())[0]
+    assert pm.manifest
+    data = json.loads(pm.manifest.json(exclude_unset=True))
+    data["display_name"] = display_name
```

Comment:
this feels like a lot of trouble and mocking to go through just to test this validator...  how about:

```py
field = PluginManifest.__fields__['display_name']
value, err = field.validate(display_name, {}, loc=0)
assert err is not None
```